cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java

```diff
@@ -87,12 +88,17 @@ public FactTable getOrCreateFactTable(int resolution) {
     }
     // for efficient counters
     props.setReadlessIncrementSupport(true);
-    // configuring pre-splits
-    props.add(HBaseTableAdmin.PROPERTY_SPLITS,
-              GSON.toJson(FactTable.getSplits(DefaultMetricStore.AGGREGATIONS.size())));
-    // Disable auto split
-    props.add(HBaseTableAdmin.SPLIT_POLICY,
-              cConf.get(Constants.Metrics.METRICS_TABLE_HBASE_SPLIT_POLICY));
+
+    // Only setup HBase specific properties if the table is backed by HBase
+    // This is a bit hacky, but is for avoiding loading HBase classes in non-hbase environment.
+    if (metricsTableDefinition instanceof HBaseMetricsTableDefinition) {
```

Comment:
Can we just use `LocalMetricsDatasetFactory` rather than adding  `instanceOf(s)`?

Reply:
Where do we use it? The definition comes from guice.

Reply:
Ideally we shouldn't have the `LocalMetricsDatasetFactory`. That factory class itself is also a hack.

Reply:
I refactored the code and guice module to remove the use of HBase classes in this factory. Also removed the LocalMetricsDatasetFactory class.